syntax = "proto3";

package com.gmail.twitte.harold;

service ExampleService {
  rpc GetValue (GetValueRequest) returns (GetValueReply);
  rpc SetValue (SetValueRequest) returns (SetValueReply);
  rpc Shutdown (ShutdownRequest) returns (ShutdownReply);
}

message GetValueRequest {
  string key = 1;
}

message GetValueReply {
  string value = 1;
}

message SetValueRequest {
  string key = 1;
  string value = 2;
}

message SetValueReply {
  bool success = 1;
}

message ShutdownRequest {}
message ShutdownReply {
  bool success = 1;
}